realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -114,6 +114,22 @@ public boolean isAutoRefresh() {
     }
 
     /**
+     * Refreshes the Realm instance and all the RealmResults and RealmObjects instances coming from it.
+     * It also calls the listeners associated to the Realm instance.
+     *
+     * Calling this on a thread with async queries will turn those queries into synchronous queries.
+     *
+     * @throws IllegalStateException if attempting to refresh from within a transaction.
+     */
+    public void refresh() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot refresh a Realm instance inside a transaction.");
```

Comment:
I think Object Store will throw this anyway? It would be no need to do it here .